src/Pages/SinglePostPage/SinglePostPageComponents/CommentSection/CommentSection.jsx

```diff
@@ -0,0 +1,65 @@
+import { useAuthContext, usePostContext } from "Context";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import styles from "./CommentSection.module.css";
+
+export const CommentSection = () => {
+  const { userState } = useAuthContext();
+  const { comments, addComment } = usePostContext();
+  const [comment, setComment] = useState({
+    firstName: userState.firstName,
+    username: userState.username,
+    comment: "",
+  });
+
+  return (
+    <>
+      <div className={styles.comment_section}>
+        <div className={styles.comment}>
+          <img src={userState.profilePhoto} alt="user-image" />
+          <textarea
+            onChange={(e) =>
+              setComment({ ...comment, comment: e.target.value })
+            }
+            placeholder="Reply Here...."
+            cols="10"
+            rows="2"
+            value={comment.comment}
+          ></textarea>
+          <button
+            onClick={() => {
+              addComment(comment);
+              setComment({ ...comment, comment: "" });
+            }}
+          >
+            Reply
+          </button>
+        </div>
+      </div>
+      {comments.length !== 0 &&
+        comments.map((commentData) => {
```

Comment:
you can try optional chaining here `?.`, so you won't have to check the length seperately.

Reply:
Thanks for pointing it out i will check it out